Replace request with got

philnash opened the pull request:
[Request is deprecated](https://www.npmjs.com/package/request) so this project should move on.

This pull request replaces request with [got](https://www.npmjs.com/package/got) which is a popular HTTP client that supports promises and streams (as used in `download.js`). I feel the result is actually simpler, which is nice, and has more deletions than additions as well as fewer dependencies.

As part of this, I did adjust a couple of things, such as:

* Rather than expose the underlying http client (previously the request client and now the got client) wrap up an `NgrokClient` class with the available API methods and expose that.
* Use `console.error` or `process.stderr` for most logging in `download.js` (mainly based on [reading this](https://www.jstorimer.com/blogs/workingwithcode/7766119-when-to-use-stderr-instead-of-stdout)).
* Removed some unused requires

This is a breaking change and will require a new version. I also have not done anything with the embedded types, which are now out of date with this code.

I hope you are interested in this work. I understand that I chose a new dependency and did this all myself without raising an issue, so if you have a HTTP client that you would prefer to use or even completely different plans, then I do not mind.

Let me know if you have any thoughts or questions. Cheers!

bubenshchykov commented on index.js:
```diff
@@ -14,7 +14,7 @@ async function connect (opts) {
 
   processUrl = await getProcess(opts);
-  internalApi = request.defaults({baseUrl: processUrl});
+  internalApi = new NgrokClient(processUrl);
```
would apiClient be a better name?

we can also mention it in readme

bubenshchykov commented on index.js:
```diff
@@ -49,7 +49,10 @@ async function connectRetry (opts, retryCount = 0) {
     if (!isRetriable(err) || retryCount >= 100) {
-      throw err.error || err.response;
+      if (err.response) {
+        throw JSON.parse(err.response.body);
```
please, wrap it into "new Error"

bubenshchykov commented:
Hi Phil, thank you so much for doing it!

- NgrokClient is awesome
- got it cool to

Thanks for sharing `console.error` thing. It looks weird and I need to wrap my head around it. 

Is it widely adopted by community? If no - just add a comment with a link at the top of download.js, and submit the PR!

philnash commented:
Hey @bubenshchykov,

Sorry I dropped this for a while, I was concerned about such a big change and just pushed it to the back of my mind.

I have returned to it though and rebased this branch with the latest from master. I know I have release rights, but I wanted to pass this by you first to make sure you're on board.

I want to release a version 4.0.0.beta.1 to the `next` tag. This will allow me to test the project out in [my VSCode extension](https://github.com/philnash/ngrok-for-vscode) and get other users testing it too, to make sure it is ready for release. That way I can fix anything that went wrong before pushing the new v4 to the `latest` tag.

Are you ok with this if I do this over the next few days?

Cheers!

bubenshchykov commented:
Hey @philnash , sorry I missed it. I'm totally Ok, thanks for the heads up!

philnash commented:
@bubenshchykov Just to let you know, it's taken me a while, but now I believe I'm ready to fully release version 4.

Please let me know if you are good with this, if I should just go ahead or if there's any last minute changes you'd make.

Thanks!
